include/swift/Basic/ExponentialGrowthAppendingBinaryByteStream.h

```diff
@@ -57,16 +57,11 @@ class ExponentialGrowthAppendingBinaryByteStream
 
   llvm::Error writeBytes(uint32_t Offset, ArrayRef<uint8_t> Buffer) override;
 
-  /// This is an optimized version of \c writeBytes that assumes we know the
-  /// size of \p Value at compile time (which in particular holds for integers).
-  /// It does so by exposing the memcpy to the optimizer along with the size 
-  /// of the value being assigned; the compiler can then optimize the memcpy
-  /// into a fixed set of instructions.
-  /// This assumes that the endianess of this steam is the same as the native
-  /// endianess on the executing machine. No endianess transformations are
-  /// performed.
```

Comment:
This comment is what I mean, but on second thought in the worst case it's byte-swapping a single fixed-size value, which is probably not that expensive.